keyboards/novelkeys/nk87/keymaps/via/keymap.c

```diff
@@ -49,3 +56,39 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
     KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS,                                              KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
 };
 
+/* Indicator LEDS are part of the LED driver
+ * Top LED is blue only. LED driver 2 RGB 63 Blue channel
+ * Middle LED is blue and red. LED driver 2 RGB 63 Red and Green channel
+ * Bottom LED is red only LED driver 2 RGB 48 Blue channel.
+ */
+uint8_t CAPS = 0;
+uint8_t FN1 = 0;
+uint8_t FN2 = 0;
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res) {
+        if (led_state.caps_lock) {
+            CAPS = 255;
+        } else {
+            CAPS = 0;
+        }
+    }
+    IS31FL3733_set_color( 63+64-1, FN1, FN1, CAPS );
+    IS31FL3733_set_color( 48+64-1, 0, 0, FN2 );
+    return res;
+}
+
+__attribute__((weak)) layer_state_t layer_state_set_user(layer_state_t state) {
+    if (state & (1UL << 1)) {
+        FN1 = 255;
+    } else {
+        FN1 = 0;
+    }
+    if (state & (1UL << 2)) {
+        FN2 = 255;
+    } else {
+        FN2 = 0;
+    }
+  return state;
```

Comment:
This is probably more readable.
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
  switch (get_highest_layer(state)) {
    case 1:
      FN1 = 255;
      break;
    case 2:
      FN2 = 255;
      break;
    default:
      FN1 = 0;
      FN2 = 0;
  }
  return state;
```

Reply:
Fixed with eab35ec7b36b75cfa5cbb1a4d9274236f7757dc7